In [ ]:
import pandas as pd
import numpy as np

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [ ]:
path = '/content/gdrive/MyDrive/Class/train.csv'
df = pd.read_csv(path, delimiter=',')
df

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]

##1. What is the comparison between passengers that Survived and Not Survived ?
##if Survived = 1
##if Not Surived = 0

##Passenger Survived


In [ ]:
Selamat = df[df.Survived ==1]
Selamat

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
8              9         1       3  ...  11.1333   NaN         S
9             10         1       2  ...  30.0708   NaN         C
..           ...       ...     ...  ...      ...   ...       ...
875          876         1       3  ...   7.2250   NaN         C
879          880         1       1  ...  83.1583   C50         C
880          881         1       2  ...  26.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
889          890         1       1  ...  30.0000  C148         C

[342 rows x 12 columns]

In [ ]:
Selamat['Survived'].sum()

342


There are 342 Survived passengers overall

##Passenger Not Survived

In [ ]:
Tidak_Selamat = df[df.Survived ==0]
Tidak_Selamat

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
4              5         0       3  ...   8.0500   NaN         S
5              6         0       3  ...   8.4583   NaN         Q
6              7         0       1  ...  51.8625   E46         S
7              8         0       3  ...  21.0750   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
884          885         0       3  ...   7.0500   NaN         S
885          886         0       3  ...  29.1250   NaN         Q
886          887         0       2  ...  13.0000   NaN         S
888          889         0       3  ...  23.4500   NaN         S
890          891         0       3  ...   7.7500   NaN         Q

[549 rows x 12 columns]

In [ ]:
Tidak_Selamat['Survived'].shape

(549,)

There are 549 Not Survived passengers overall

##2. How many Survived passengers from each pclass ?


Total Survived in Pclass 1

In [ ]:
Survived1 =df[(df.Survived == 1) & (df.Pclass == 1)]
Survived1['Total_Survived'] = Survived1['Survived'].sum()
Survived1

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


PassengerId  Survived  Pclass  ... Cabin Embarked  Total_Survived
1              2         1       1  ...   C85        C             136
3              4         1       1  ...  C123        S             136
11            12         1       1  ...  C103        S             136
23            24         1       1  ...    A6        S             136
31            32         1       1  ...   B78        C             136
..           ...       ...     ...  ...   ...      ...             ...
862          863         1       1  ...   D17        S             136
871          872         1       1  ...   D35        S             136
879          880         1       1  ...   C50        C             136
887          888         1       1  ...   B42        S             136
889          890         1       1  ...  C148        C             136

[136 rows x 13 columns]

Total Survived in Pclass 2

In [ ]:
Survived2 =df[(df.Survived == 1) & (df.Pclass == 2)]
Survived2['Total_Survived'] = Survived2['Survived'].sum()
Survived2

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


PassengerId  Survived  Pclass  ... Cabin Embarked  Total_Survived
9             10         1       2  ...   NaN        C              87
15            16         1       2  ...   NaN        S              87
17            18         1       2  ...   NaN        S              87
21            22         1       2  ...   D56        S              87
43            44         1       2  ...   NaN        C              87
..           ...       ...     ...  ...   ...      ...             ...
831          832         1       2  ...   NaN        S              87
865          866         1       2  ...   NaN        S              87
866          867         1       2  ...   NaN        C              87
874          875         1       2  ...   NaN        C              87
880          881         1       2  ...   NaN        S              87

[87 rows x 13 columns]

Total Survived in Pclass 3

In [ ]:
Survived3 =df[(df.Survived == 1) & (df.Pclass == 3)]
Survived3['Total_Survived'] = Survived3['Survived'].sum()
Survived3

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


PassengerId  Survived  Pclass  ... Cabin Embarked  Total_Survived
2              3         1       3  ...   NaN        S             119
8              9         1       3  ...   NaN        S             119
10            11         1       3  ...    G6        S             119
19            20         1       3  ...   NaN        C             119
22            23         1       3  ...   NaN        Q             119
..           ...       ...     ...  ...   ...      ...             ...
838          839         1       3  ...   NaN        S             119
855          856         1       3  ...   NaN        S             119
858          859         1       3  ...   NaN        C             119
869          870         1       3  ...   NaN        S             119
875          876         1       3  ...   NaN        C             119

[119 rows x 13 columns]

##3. What is the average and median age of passengers that Survived and Not Survived ?

In [ ]:
Selamat['Age'].agg([np.mean, np.median])

mean      28.34369
median    28.00000
Name: Age, dtype: float64

In [ ]:
Tidak_Selamat['Age'].agg([np.mean, np.median])

mean      30.626179
median    28.000000
Name: Age, dtype: float64

##4. Create a new column named “age_interval”

In [ ]:
df.loc[ (df.Age < 25), 'Age_Interval'] = '< 25'
df.loc[ (df.Age >= 25) & (df.Age <= 35 ), 'Age_Interval'] = '25 - 35'
df.loc[ (df.Age > 35) & (df.Age < 51 ), 'Age_Interval'] = '36 - 50'
df.loc[ (df.Age > 50), 'Age_Interval'] = '> 50'

df

PassengerId  Survived  Pclass  ... Cabin Embarked  Age_Interval
0              1         0       3  ...   NaN        S          < 25
1              2         1       1  ...   C85        C       36 - 50
2              3         1       3  ...   NaN        S       25 - 35
3              4         1       1  ...  C123        S       25 - 35
4              5         0       3  ...   NaN        S       25 - 35
..           ...       ...     ...  ...   ...      ...           ...
886          887         0       2  ...   NaN        S       25 - 35
887          888         1       1  ...   B42        S          < 25
888          889         0       3  ...   NaN        S           NaN
889          890         1       1  ...  C148        C       25 - 35
890          891         0       3  ...   NaN        Q       25 - 35

[891 rows x 13 columns]

##5. After creating the new column in question number 4, which age interval has the highest percentage of Survived ?

In [ ]:
Percent_Survived = df.groupby('Age_Interval').sum()
Percent_Survived

PassengerId  Survived  Pclass      Age  SibSp  Parch       Fare
Age_Interval                                                                 
25 - 35            101017        89     506  6514.50     72     40  6557.6580
36 - 50             71044        61     294  6436.50     59     82  6750.0917
< 25               119819       118     698  4496.17    219    167  8673.4249
> 50                28408        22      99  3758.00     16     19  2790.7084


##Seen in the table above the most surviving is Age_Interval < 25 with a total of 118 passengers, which means 7,55% from all passenger from all class. 13,243%